functions/src/firebase-delete-collection.ts

```diff
@@ -0,0 +1,67 @@
+//From firebase documentation: https://firebase.google.com/docs/firestore/manage-data/delete-data#collections
+
+export const deleteCollection = async (
+  fireStore: FirebaseFirestore.Firestore,
+  collectionPath: string,
+  batchSize: number
+) => {
+  const collectionRef = fireStore.collection(collectionPath);
+  const query = collectionRef.limit(batchSize);
+
+  return new Promise((resolve, reject) => {
+    deleteQueryBatch(fireStore, query, batchSize, resolve, reject).catch(
+      reject
+    );
+  });
+};
+
+async function deleteQueryBatch(
+  fireStore: FirebaseFirestore.Firestore,
+  query: FirebaseFirestore.Query,
+  batchSize: number,
+  resolve,
+  reject
+) {
+  query
+    .get()
+    .then(async snapshot => {
+      // When there are no documents left, we are done
+      if (snapshot.size === 0) {
+        return 0;
+      }
+      // Delete documents in a batch
+      const batch = fireStore.batch();
+      for (const doc of snapshot.docs) {
+        const collections = await doc.ref.getCollections();
+        collections.forEach(async collection => {
+          await deleteCollection(fireStore, collection.path, 100).catch(reject);
+        });
+      }
+      snapshot.docs.forEach(doc => {
+        batch.delete(doc.ref);
+      });
+
+      return batch.commit().then(() => {
+        return snapshot.size;
+      });
+    })
+    .then(numDeleted => {
+      if (numDeleted === 0) {
+        resolve();
+        return;
+      }
+
+      // Recurse on the next process tick, to avoid
+      // exploding the stack.
+      process.nextTick(async () => {
+        await deleteQueryBatch(
+          fireStore,
+          query,
+          batchSize,
+          resolve,
+          reject
+        ).catch(reject);
+      });
+    })
+    .catch(reject);
```

Comment:
According to the documentation, this code is a simplified example and is not recommended for production (also it is not good code, quality-wise). The doc instead recommends to use `firebase-tools`. See [here](https://github.com/firebase/firebase-tools/blob/master/lib/firestore/delete.js) for the code and [here](https://github.com/firebase/firebase-tools/blob/master/commands/firestore-delete.js#L78) for how to call it.